src/utils/SdkCoin.sol

```diff
@@ -44,42 +44,57 @@ library SdkCoin {
      * @return convertedAmount The amount converted to uint64 supported by cosmos coins
      * @return remainder The remainder of the conversion
      */
-    function _convertERC20AmountToSdkCoin(
+    function _ERC20ToSdkCoin_ConvertAmount(
         address tokenAddress,
         uint256 amount
     )
         internal
         view
         returns (uint64, uint256)
     {
-        // Note if we want to allow different cosmos decimals, we need to handle that here too
-        // Note that tokenAddress input validations are executed in the _getERC20TokenDecimals function
+        // Retrieve the number of decimals the ERC20 token uses
+        // Note that input validations for the tokenAddress are performed within _getERC20TokenDecimals
         uint8 tokenDecimals = _getERC20TokenDecimals(tokenAddress);
 
-        if (tokenDecimals < 6) {
+        // Ensure the token's decimals are within the supported range
+        if (tokenDecimals < 6 || tokenDecimals > 77) {
             revert IISdkCoinErrors.UnsupportedTokenDecimals(tokenDecimals);
         }
-        // Amount input validation
+
+        // Ensure the provided amount is not zero, as zero is an invalid amount
         if (amount == 0) {
             revert IISdkCoinErrors.InvalidAmount(amount);
         }
-        // Ensure the amount respects the token's decimals
-        // Handle the case where the input amount exceeds the token's precision
-        uint256 tempConvertedAmount;
+
+        // Note that for values of amount > ~uint64(0) which is the max value of uint64 type
+        // The function will revert automatically for the safeCast conversion function. No need to check it here.
+
+        // Variables to store the converted amount and any remainder
+        uint256 temp_convertedAmount;
```

Comment:
The name of this variable was changed to `tempConvertedAmount` due to linter warning. I will force warnings to be errors now to make sure we see these in the CI.

Also, it seems you haven't merged main to this branch properly, which is why I had to push a fix commit. You preferred to keep the changes in this branch instead of main (which was ahead of this branch). Next time, you should prefer main and make any changes needed in your branch to make it compatible with main. (Instead of reverting main)

Reply:
Very true, I fixed conflicts and not merged main into the branch. My bad.